### ISSUE TITLE
systemd-udevd: DEVPATH and INTERFACE properties are not updated when network device name is changed more than one time

### ISSUE THREAD

**Issue by keecin** [systemd/systemd#9426]
**systemd version the issue has been seen with**
>V238

**Used distribution**
 >Fedora 28

**Expected behaviour you didn't see**
>DEVPATH and INTERFACE properties are not updated when network device name is changed more than one time

**Unexpected behaviour you saw**
 > DEVPATH and INTERFACE properties are not updated when network device name is changed more than one time

**Steps to reproduce the problem**
1. create a virtual network device named veth0 and show properties of DEVPATH and INTERFACE
`ip link add name veth0 type veth peer name veth1`
`udevadm info -p /sys/devices/virtual/net/veth0 | grep -e INTERFACE -e DEVPATH`
output is veht0 as what we expected
>E: DEVPATH=/devices/virtual/net/veth0
> E: INTERFACE=veth0

2. rename veth0 to veth2
`ip link set dev veth0 name veth2`
`udevadm info -p /sys/devices/virtual/net/veth2 | grep -e INTERFACE -e DEVPATH`
output is veht2 as what we expected
 >E: DEVPATH=/devices/virtual/net/veth2
 >E: INTERFACE=veth2

3. rename veth2 to veth3
`ip link set dev veth2 name veth3`
`udevadm info -p /sys/devices/virtual/net/veth3 | grep -e INTERFACE -e DEVPATH`
this time output is veth2, not veth3
 >E: DEVPATH=/devices/virtual/net/veth2
 >E: INTERFACE=veth2

4. rename veth3 to veth4
`ip link set dev veth3 name veth4`
`udevadm info -p /sys/devices/virtual/net/veth3 | grep -e INTERFACE -e DEVPATH`
output is still veth2, not veth4
 >E: DEVPATH=/devices/virtual/net/veth2
 >E: INTERFACE=veth2

The full script for the above steps.
`ip link add name veth0 type veth peer name veth1
udevadm info -p /sys/devices/virtual/net/veth0 | grep -e INTERFACE -e DEVPATH

sleep 1
ip link set dev veth0 name veth2
echo "after changing to veth2"
udevadm info -p /sys/devices/virtual/net/veth2 | grep -e INTERFACE -e DEVPATH

sleep 1
echo "after changing to veth3"
ip link set dev veth2 name veth3
udevadm info -p /sys/devices/virtual/net/veth3 | grep -e INTERFACE -e DEVPATH

sleep 1
echo "after changing to veth4"
ip link set dev veth3 name veth4
udevadm info -p /sys/devices/virtual/net/veth4 | grep -e INTERFACE -e DEVPATH`

This problem can also be reproduced on V219 and V239

**Comment by arvidjaar** [comment]
Likely duplicate of issue 9006.

**Comment by poettering** [comment]
@keecin hmm, do you see udev events being generated for these events? does "udevadm monitor" show the renames properly?

**Comment by keecin** [comment]
@poettering

from the second rename, DEVPATH and INTERFACE always shown as "veth2" from "udevadm monitor -p"

The output from first rename(veth0->veth2) is below.
```
ACTION=move
DEVPATH=/devices/virtual/net/veth2
DEVPATH_OLD=/devices/virtual/net/veth0
IFINDEX=8
INTERFACE=veth2
SEQNUM=2437
SUBSYSTEM=net

UDEV  [7635.858142] move     /devices/virtual/net/veth2 (net)
ACTION=move
DEVPATH=/devices/virtual/net/veth2
DEVPATH_OLD=/devices/virtual/net/veth0
ID_NET_DRIVER=veth
IFINDEX=8
INTERFACE=veth2
SEQNUM=2437
SUBSYSTEM=net
SYSTEMD_ALIAS=/sys/subsystem/net/devices/veth0 /sys/subsystem/net/devices/veth2
TAGS=:systemd:
UDEV_BIOSDEVNAME=0
USEC_INITIALIZED=23027908
biosdevname=0
net.ifnames=0
```

The output from second rename(veth2->veth3) is below, this time it is still shown as eth2.
```
ACTION=move
DEVPATH=/devices/virtual/net/veth3
DEVPATH_OLD=/devices/virtual/net/veth2
IFINDEX=8
INTERFACE=veth3
SEQNUM=2438
SUBSYSTEM=net

UDEV  [7661.882324] move     /devices/virtual/net/veth2 (net)
ACTION=move
**DEVPATH=/devices/virtual/net/veth2**
DEVPATH_OLD=/devices/virtual/net/veth2
ID_NET_DRIVER=veth
IFINDEX=8
**INTERFACE=veth2**
SEQNUM=2438
SUBSYSTEM=net
SYSTEMD_ALIAS=/sys/subsystem/net/devices/veth0 /sys/subsystem/net/devices/veth2 /sys/subsystem/net/devices/veth3
TAGS=:systemd:
UDEV_BIOSDEVNAME=0
USEC_INITIALIZED=23027908
biosdevname=0
net.ifnames=0
```

**Comment by yuwata** [comment]
Fix is waiting in #11501. PTAL.
